cairo_programs/bad_programs/bad_usort.cairo

```diff
@@ -0,0 +1,113 @@
+%builtins range_check
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.math import assert_lt, assert_nn
+
+# Sorts an array of field elements and removes duplicates.
+# Returns the sorted array and an array of multiplicities.
+# multiplicities[i] is the number of times that output[i] appeared in input.
```

Comment:
Why do we need to check this?
What we really need to check is:
```
for i in range(0, len(output)): output[i] in input
for j in range(0, len(input)): input[j] in output
for k in range(1, len(output)): output[k-1] < output[k]
```

That's all the invariant for the function, right?

Reply:
Or is it part of the hint itself?

Reply:
Yes, it's part of the hint and I'm only implementing it like this to check that one of the hints in verify multiplicities works